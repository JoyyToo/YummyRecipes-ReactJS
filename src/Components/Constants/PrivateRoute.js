import React from 'react';
import { Route, Redirect } from 'react-router-dom';

// function for private urls which cant be accessed before login
const PrivateRoute = ({ component: Values, Component, ...rest }) => (
    <Route
    {...rest}
    render={props => (
          window.localStorage.getItem('token') ? 
          (<Values {...props} />) :
           (<Redirect to={{ pathname: '/login' }} />
        ))}
  />
);
export default PrivateRoute;