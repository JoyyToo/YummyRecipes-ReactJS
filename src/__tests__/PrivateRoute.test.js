import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PrivateRoute from '../Components/Constants/PrivateRoute';

describe('PrivateRoute component', () => {
  const wrapper = shallow(<PrivateRoute />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
