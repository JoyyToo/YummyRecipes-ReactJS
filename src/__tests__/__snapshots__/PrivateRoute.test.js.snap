// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component renders properly 1`] = `
<Route
  render={[Function]}
/>
`;
